src/jobflow_remote/jobs/upgrade.py

```diff
@@ -37,19 +37,44 @@ class UpgradeAction:
     required: bool = False
 
 
+@dataclass
+class UpgradeCondition:
+    """Details of an upgrade condition"""
+
+    description: str
+    collection: str
+    query: dict
+    one_doc_check: Callable | None = None
+
+    def check(self, job_controller: JobController) -> dict | None:
+        collection = getattr(job_controller, self.collection)
+        if collection is None:
+            return None
+        count = collection.count_documents(self.query)
+        if count == 0:
+            return None
+        if count == 1 and self.one_doc_check is not None:
+            doc = collection.find_one(self.query)
+            count = self.one_doc_check(doc)
+            if count == 0:
+                return None
+        return {"condition": self, "count": count}
```

Comment:
I suppose the object can also change in the future, but it seems to handle a limited number of checks. I wonder if it would be easier that the `UpgradeCondition` only has a `description` and an abstract `check` method? Maybe one case make a subclass that implements potential standard checks like `NoDocuments` with also a `collection`+ `query` argument. The `one_doc_check` is not so different and it seems to make the logic less transparent.
If you think it is not worth, it can also be left for the future, altough it may be annoying to refactor this later on.